modules/forms_api/app/services/forms_api/pdf_stamper.rb

```diff
@@ -69,6 +69,27 @@ def self.stamp2110210(generated_form_path, data)
       Common::FileHelpers.delete_file_if_exists(stamp_path) if defined?(stamp_path)
     end
 
+    def self.stamp21p0847(generated_form_path, data)
+      desired_stamps = [[50, 190]]
+      first_name = data.dig('preparer_name', 'first')
+      middle_name = data.dig('preparer_name', 'middle')
+      last_name = data.dig('preparer_name', 'last')
+      signature_text = "#{first_name} #{middle_name} #{last_name}"
+
+      stamp_path = Common::FileHelpers.random_file_path
+      Prawn::Document.generate(stamp_path, margin: [0, 0]) do |pdf|
+        pdf.start_new_page
+        pdf.draw_text signature_text, at: desired_stamps[0], size: 16
+      end
+
+      multistamp(generated_form_path, stamp_path)
+    rescue
+      Rails.logger.error "Failed to generate stamped file: #{e.message}"
```

Comment:
I don't think this is going to work unless you make it `rescue => e`. And I'd really recommend capturing specific exceptions rather than any exception, but I won't block you on that.

Reply:
and maybe test exception handling if you're going to leave it so open ended?

Reply:
Thanks for calling that out! I refactored the stampers to cut down on that reused logic, added the ` => e` and added tests for errors.

Reply:
I still left the error generic because there are a few failure points, and which one it is isn't important for us right now. I will keep that suggestion in mind.

Reply:
It's totally your call